tests/raftstore_cases/test_region_heartbeat.rs

```diff
@@ -153,3 +153,31 @@ fn test_server_pending_peers() {
     let mut cluster = new_server_cluster(0, 3);
     test_pending_peers(&mut cluster);
 }
+
+#[test]
+fn test_region_heartbeat_cache() {
+    let mut cluster = new_server_cluster(0, 1);
+    let heartbeat_interval_ms = 50;
+    let heartbeat_ttl = 9;
+
+    cluster.cfg.raft_store.pd_heartbeat_tick_interval =
+        ReadableDuration::millis(heartbeat_interval_ms);
+    let pd_client = Arc::clone(&cluster.pd_client);
+    cluster.run();
+
+    // Write the initial value for a key, so it will heartbeat pd.
+    let key = b"k";
+    cluster.must_put(key, b"v1");
+    sleep(Duration::from_millis(heartbeat_interval_ms));
+    let count = pd_client.get_region_heartbeat_count();
+
+    // No heartbeat during ttl.
+    sleep(Duration::from_millis(
+        heartbeat_interval_ms * heartbeat_ttl / 3,
+    ));
+    assert_eq!(pd_client.get_region_heartbeat_count(), count);
+
+    // After ttl, there must be exact one heartbeat.
+    sleep(Duration::from_millis(heartbeat_interval_ms * heartbeat_ttl));
+    assert_eq!(pd_client.get_region_heartbeat_count(), count + 1);
+}
```

Comment:
I think we need a test that heartbeat is sent when ttl is not expired and heartbeat is updated.